addons/mail/static/src/components/attachment_box/attachment_box.js

```diff
@@ -86,6 +86,8 @@ class AttachmentBox extends Component {
     _onAttachmentRemoved(ev) {
         // FIXME Could be changed by spying attachments count (task-2252858)
         this.trigger('o-attachments-changed');
+        // when an attachment is deleted from attachment box, the form view need to reload.
+        this.trigger('reload');
```

Comment:
Hello, you should set `post_refresh` `always` on the chatter field of view that requires this behavior.
It will catch `o-attachments-changed` and trigger the reload properly.
However I've heard there was a bug with it, so feel free to test it out and let me know/fix it if it doesn't work as intended.

We don't want to reload views in general, this should be view specific.